synapse/lib/stormlib/infosec.py

```diff
@@ -409,6 +409,72 @@ def CVSS3_1_calc(vdict):
 def roundup(x):
     return (math.ceil(x * 10) / 10.0)
 
+@s_stormtypes.registry.registerLib
+class MitreAttackFlowLib(s_stormtypes.Lib):
+    '''
+    A Storm library which implements modeling MITRE ATT&CK Flow diagrams.
+    '''
+    _storm_lib_path = ('infosec', 'mitre', 'attack', 'flow')
+    _storm_locals = (
+        {'name': 'ingest', 'desc': 'Ingest a MITRE ATT&CK Flow diagram in JSON format.',
+         'type': {'type': 'function', '_funcname': '_storm_query',
+                  'args': (
+                      {'name': 'flow', 'type': 'data', 'desc': 'The JSON data to ingest.'},
+                  ),
+                  'returns': {'type': 'null'}}},
+
+    )
+    _storm_query = '''
+        function ingest(flow) {
+            $lib.cast(it:mitre:attack:flow:json, $flow)
+
+            $guid = $lib.guid($flow)
```

Comment:
Are thse JSON blobs pretty stable?  I'm wondering if we end up having some order dependency in this guid grinding.  Did you consider a normalization step, or using a unique id within the JSON obj if it's there?

Reply:
Actually, running it through schema validation might sort it, if so you do $flow=$lib.cast(...), but the question still remains whether the entire json blob needs to be identiical in order to deconflict on an it:mitre:attack:flow node.

Reply:
It might be worthwhile to add a `flatten=(false)` default argument to `$lib.guid()` and have it pass through synapse.common.flatten if specified, that way we avoid dict order deps.

Reply:
I made a `$lib.infosec.mitre.attack.flow.norm()` function which does a flatten and then sorts the dicts in the `objects` list by the `id` key.